data_validation/src/spark_rapids_validation_tool/validation_scripts/metadata_validation.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from pyspark import SparkContext
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, min, max, avg, stddev, countDistinct, when, asc, round
+import fnmatch
+
+def validation(spark, args):
+
+    if not valid_input(spark,args):
+        print('|--Please Check The Inputs --|')
+        return
+
+    result = top_level_metadata(spark, args.format, args.t1, args.t2, args.t1p, args.t2p, args.f)
+    print('|--Top Level Metadata Info--|')
+    print(result.show())
+
+    result = metrics_metadata(spark, args.format, args.t1, args.t2, args.t1p, args.t2p, args.pk, args.i, args.e, args.f, args.p)
+    if result.count() == 0:
+        print(f'|--Table {args.t1} and Table {args.t2} has identical metadata info--|')
+        print(result.show())
+    else:
+        print('|--Metadata Diff Info--|')
+        print(result.show())
+
+    print('|--Run Metadata Validation Success--|')
+
+def valid_input(spark, args):
+    """
+    Check the input is valida for matadata validation tool
+    1- valid table
+    2- valid included column
+    3- check format supported
+    """
+    if not valid_table(spark, args):
+        return False
+    if not valid_metadata_included_column(spark, args):
+        return False
+    if args.format != 'hive':
+        print('|--Currently only support hive format--|')
+    return True
+
+def valid_table(spark, args):
+    """
+    Check if the tables exist
+    """
+    if not spark._jsparkSession.catalog().tableExists(args.t1):
+        print(f'|--Table {args.t1} does not exist!--|')
+        return False
+    if not spark._jsparkSession.catalog().tableExists(args.t2):
+        print(f'|--Table {args.t2} does not exist!--|')
+        return False
+    return True
+
+def valid_metadata_included_column(spark, args):
+    """
+    Check if the included column valid
+    """
+    if args.i in ['None', 'all']:
+        return True
+    table_DF = load_table(spark, args.format, args.t1, args.t1p, args.pk, args.e, args.i, args.f, "")
+    excluded_columns_list = [e.strip() for e in args.e.split(",")]
+    verify_column = [i.strip() for i in args.i.split(",") if i not in excluded_columns_list]
+    verify_DF = table_DF.select(verify_column)
+
+    for c in verify_DF.schema.fields:
+        # here only excluded 'data' because it will raise exception, we also should excluded str/map/nested
```

Comment:
Chanfe 'data' to 'date'. 
Is there a follow on issue for excluding string/map/nested ? Curious why we are excluding string?

Reply:
I am not sure about which should be excluded for now, maybe we can keep it there and wait for the requirement from customers.